### PR TITLE
Feature/mobilenetv2 bitnet

### PR DESCRIPTION
**Pull request by ayeganov**
# What Does This PR Do?

Introducing an example of running mobilenet 2 with bitnet linear and conv2d layers.

### CODE REVIEW

File: examples/mobilenet2_example.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+def main():
+    num_classes: int        = 100
+    learning_rate: float    = 1e-3
+    num_epochs: int         = 10
+    batch_size: int         = 128
+
+    transform = transforms.Compose([
+        transforms.Resize((224, 224)),
```
Comment (dariocazzani) [inline examples/mobilenet2_example.py]: CIFAR has an input size of 32x32, why are you resizing?
That's why it trains so slowly

File: examples/mobilenet2_example.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+from torch import nn
+from torchvision import transforms, datasets
+from torch.utils.data import DataLoader
+
+from tqdm import tqdm
+
+from bitnet.nn.bitlinear import BitLinear
+from bitnet.nn.bitconv2d import BitConv2d
+from bitnet.models.resnet import resnet18
+from bitnet.models.mobilenet2 import create_mobilenet2, create_bit_mobilenet2
```
Comment (dariocazzani) [inline examples/mobilenet2_example.py]: Not a fan of the `create` functions.
I wanted to keep the interface as similar as possible to the "standard" ones.

File: examples/mobilenet2_example.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import torch
+from torch import nn
+from torchvision import transforms, datasets
+from torch.utils.data import DataLoader
+
+from tqdm import tqdm
+
+from bitnet.nn.bitlinear import BitLinear
+from bitnet.nn.bitconv2d import BitConv2d
+from bitnet.models.resnet import resnet18
```
Comment (dariocazzani) [inline examples/mobilenet2_example.py]: Dead import

**Review by dariocazzani — CHANGES_REQUESTED** [review]

File: examples/mobilenet2_example.py (hunk 1 above)
Comment (ayeganov) [reply examples/mobilenet2_example.py]: Because MNV2 was meant for that size - if I don't resize the accuracy drops down from 55% to 19% with BitNet and 61% to 27% for FloatNet. Do you think the drop in accuracy is acceptable?

File: bitnet/models/mobilenet2.py [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+            norm_layer,
+            activation_layer,
+            dilation,
+            inplace,
+            bias,
+            torch.nn.Conv2d,
+        )
+
+
+class Conv2dBitNormActivation(ConvNormActivation):
```
Comment (dariocazzani) [inline bitnet/models/mobilenet2.py:155]: Not sure about this approach, now this is a copy and paste of `Conv2dNormActivation`
Why don't you want to pass a reference to either `nn.Conv2d` or `nn.BitConv2d `?

**Review by dariocazzani — CHANGES_REQUESTED** [review]

File: bitnet/models/mobilenet2.py (hunk 4 above)
Comment (ayeganov) [reply bitnet/models/mobilenet2.py:155]: This is a leaf implementation of the base ConvNormActivation. I am passing `BitConv2d`. This class clearly indicates its type.

**Review by dariocazzani — APPROVED** [review]
